pkg/operator/controller/ingress/deployment.go

```diff
@@ -820,12 +821,29 @@ func desiredRouterDeployment(ci *operatorv1.IngressController, ingressController
 				cookieName = accessLogging.HTTPCaptureCookies[0].NamePrefix
 			}
 			if maxLength == 0 {
-				maxLength = 256
+				// unless otherwise specified, haproxy sets the max cookie
+				// capture length to 63.
+				maxLength = 63
 			}
 			env = append(env, corev1.EnvVar{
 				Name:  RouterCaptureHTTPCookies,
 				Value: fmt.Sprintf("%s:%d", cookieName, maxLength),
 			})
+			// By default, captured cookies go into a datastructure that's 64
+			// bytes, with one byte reserved, so the max length of the cookie
+			// itself is 63 bytes.  If the captured length is greater than 63
+			// bytes, we can tune that data strucutre to be larger using
+			// tune.http.cookielen (exposed through the env var
+			// ROUTER_COOKIELEN). However, this is not advised since the cookie
+			// is captured for every request and every response, and the
+			// maximum length is always allocated, so the memory impact is
+			// high.
+			if maxLength > 63 {
```

Comment:
`maxLength` should be evaluated prior to setting `RouterCaptureHTTPCookies`, since it uses the `maxLength`.